packages/@react-stately/menu/src/useSubMenuTriggerState.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {FocusStrategy} from '@react-types/shared';
+import {Key, useCallback, useMemo, useState} from 'react';
+import type {MenuTreeState} from './useMenuState';
+import type {OverlayTriggerState} from '@react-stately/overlays';
+import type {TreeState} from '@react-stately/tree';
+
+export interface SubMenuTriggerProps {
+  /** Key of the trigger item. */
+  triggerKey: Key
+}
+
+export interface SubMenuTriggerState extends OverlayTriggerState {
+  // TODO descriptions
+  // Need some form of isOpen and close for usePopover. Also need closeAll for onClose for menu items, would be kinda weird
+  // to pass in both menuTreeState and the state returned by useSubMenuTriggerState to useMenu.
+  isOpen: boolean,
+  focusStrategy: FocusStrategy | null,
+  open: (focusStrategy?: FocusStrategy | null) => void,
+  close: () => void,
+  closeAll: () => void,
+  // TODO: add setOpen and toggle for type parity with useOverlayTriggerState type that Tray and Popover expect. Perhaps call useOverlayTriggerState in
+  // this hook and just make it controlled? Then spread? Kinda weird since they are unused but needed for type parity.
+  toggle: (focusStrategy?: FocusStrategy | null) => void,
+  setOpen: () => void
```

Comment:
Since this state acts as the replacement state for Popover/Tray, it needs the same type parity as `useOverlayTriggerState` and thus has toggle and setOpen even though those never get used by Tray/Popover. Don't really wanna call useOverlayTriggerState in here if I can avoid it, so I provide a no-op for setOpen

Reply:
Keep as is for now, refactor if we make Popover accept isOpen/onOpenChange instead of the entire state object of OverlayTriggerState